code/iaas/lb-logic/src/main/java/io/cattle/iaas/lb/api/action/LoadBalancerAddHostActionHandler.java

```diff
@@ -38,17 +31,8 @@ public Object perform(String name, Object obj, ApiRequest request) {
         LoadBalancer lb = (LoadBalancer) obj;
         long hostId = DataAccessor.fromMap(request.getRequestObject()).withKey(LoadBalancerConstants.FIELD_LB_HOST_ID)
                 .as(Long.class);
-        createLbHostMap(lb.getId(), hostId);
+        lbService.addHostToLoadBalancer(lb, hostId);
 
         return objectManager.reload(lb);
```

Comment:
When host is added to the load balancer, load balancer state doesn't change. And we never use the lb state to signal to start lb instance + update haproxy content. Instead the latter actions get triggered by mapping creation - host add/remove, target add/remove. So LB should always be open for modifications, and all of them are scheduled in async manner.